### ISSUE TITLE
Executing a final local or remote action after task execution

### ISSUE THREAD

**Issue by jordiclariana** [fabric/fabric#1414]
Just an idea:

It would be nice to have the possiblity to define a function that should be locally or remotely (specifying where) executed after a task has successfully finished. This function would be executed only once and only after the task has finished. Whether it is executed also if the task fails should be decided by parameters. By default only when it succeeds.

My idea is, for instance, send an email after a good deploy, or maybe set a value to graphite so I can draw an [annotation on Grafana](http://docs.grafana.org/reference/annotations/), etc.

Thanks.

**Comment by bitprophet** [comment]
This will be available in version 2, the underpinnings of which already have pre/post tasks available :) see e.g. http://www.fabfile.org/roadmap.html and http://docs.pyinvoke.org/en/0.11.1/concepts/execution.html#pre-and-post-tasks - thanks!

**Comment by jordiclariana** [comment]
Thanks @bitprophet. Meanwhile I figured out how to do itusing "execute":

```
@task
@runs_once
@hosts('host1', 'host2', 'host3')
def deploy():
    results = execute(_deploy, hosts = env.all_hosts)
    if not False in results.values():
        post_task()

@parallel
def _deploy():
    do_things()
    return(True)
```

No ideal but works.
